docs(agentic-ai): Add initial docs for AI Agent connectors

maff opened the pull request:
## Description

Provides initial documentation for the new AI Agent + Ad-hoc tools schema connectors.

Closes #5776 

<!-- Provide an overview of what to expect in the PR. -->
<!-- Relate or link the associated epic or task. -->
<!-- Add `@camunda/tech-writers` as reviewer to pull in a tech writer, or add your embedded tech writer. -->

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the Documentation team. (create draft PR and/or add `hold` label)
- [x] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [x] My changes are for **an upcoming minor release** and are in the `/docs` directory (version 8.8).
- [ ] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

- [x] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.

- [x] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [x] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

YanaSegal commented:
H @maff thanks so much for these docs, here is my feedback:
AI Agent connector 
1. `Important` gives a sense of urgency but these links are also mentioned int he additional context - so I would remove this too. 
2. I think the first paragraph is unnecessary. It makes sense in the blog post, but it is too wordy for the documents. And whatever is mentioned in this intro is explained very well in the next paragraphs. 
<img width="971" alt="image" src="https://github.com/user-attachments/assets/b46aa7c0-a8d0-4c34-bd20-e5a3be44fae5" />
3. the concept image is great! can we remove the extract sparkles? 
4. Tool calling loop paragraph: `as shown in the figure above.` please use diagram instead figure for clarity. 
5. for the configuration part - Lets reuse the pattern @mesellings created for [IDP](https://preview.docs.camunda.cloud/pr-5782/docs/next/components/modeler/web-modeler/idp/#configure-idp). Then agent as a concept can be separated a bit from the connector documentation. Happy to help.
<img width="1005" alt="image" src="https://github.com/user-attachments/assets/363d4fb4-c56b-44ad-bb25-aa4821d33071" />
6. Preconfiguration is required to use this connector, so I would like to frame it. I think framing the 2 info notifications as a prerequisite will make them clearer to the user. also this can be a place to mention Known limitations.
<img width="1007" alt="image" src="https://github.com/user-attachments/assets/ce2148da-28b2-4dc1-b2fa-59f0e6f42b15" />
7. Document section - 
7.a would it make sense to call this section "Agent document handling"? 
7.b. the 2 notifications - see image:
![image](https://github.com/user-attachments/assets/ee35abae-319e-487c-bca3-a0749463c967)
8. lets sync about the tools section - we are mixing the conceptual and the technical concepts.

In general it is very helpful documentation, I am just wondering if we should separate the concept and the technical content.

mesellings commented:
I'll try and look at this today/tomorrow - I'll make any commits directly to the branch 👍

mesellings commented:
@maff I notice the docs don't match up exactly to the AI Agent (alpha) connector in Dev - should I be using that as the 'model' for the docs? E.g.
- There is no Context section in the docs
- Guardrails in Dev, Limits in docs?

maff commented:
@mesellings as part of the following PRs the properties in the element template were shifted around/renamed a bit - in the process definition I shared, the upper (standalone) AI Agent task is the one using the current dev version (the one we should address).

- https://github.com/camunda/connectors/pull/4721
- https://github.com/camunda/connectors/pull/4768

> There is no Context section in the docs

I moved the Agent Context variable field to the "Memory" section.

> Guardrails in Dev, Limits in docs?

We initially misnamed this a bit and renamed this to limits. Guardrails are more of the overall concept including what is in the prompt vs. limits which are a hard numeric value like maximum amount of loops.

maff commented on docs/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent-example.md:
```diff
@@ -0,0 +1,236 @@
+
+### Configure multi-instance execution
+
+The ad-hoc sub-process must be configured as a [parallel multi-instance](../../modeler/bpmn/multi-instance/multi-instance.md) sub-process.
```
Technically it could be also sequential - should we rephrase it to suggest parallel?

mesellings replied:
Done

mesellings commented on docs/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent.md:
```diff
@@ -0,0 +1,297 @@
+
+The AI Agent connector enables AI agents to integrate with an LLM to provide interaction/reasoning capabilities. This connector is designed for use with an ad-hoc sub-process in a feedback loop, providing automated user interaction and tool selection.
+
+For example, use this connector to enable an AI agent to autonomously select and execute ad-hoc sub-processes by evaluating the current process context and determining the relevant tasks and tools to use in response. You can also use the AI Agent connector independently, although it is designed to be used with an ad-hoc sub-process to define the tools an AI agent can use.
```
Done

mesellings commented on docs/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent.md:
```diff
@@ -0,0 +1,297 @@
+
+| Prerequisite                                      | Description                                                                                                                                                                                                                                                                                                                                               ...
+| :------------------------------------------------ | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| Set up your LLM model provider and authentication | <p>Prior to using this connector, you must have previously set up an account with access and authentication details for the supported LLM model provider you want to use.</p><p>For example:<ul><li><p>To use an LLM model provided by Amazon Bedrock, you must have an AWS account with an access key and secret key to execute `InvokeModel` or `Conver ...
```
Done

mesellings commented on docs/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent.md:
```diff
@@ -0,0 +1,297 @@
+
+Select this option to use the [OpenAI Chat Completion API](https://platform.openai.com/docs/api-reference/chat).
+
+| Field           | Required | Description                                                                                                                                              |
```
I'll add them in for now with whatever detail I can figure out from the screenshot - we can improve the description as a follow-up if needed 👍

mesellings commented on docs/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent.md:
```diff
@@ -0,0 +1,297 @@
+
+Select the model you want to use for the selected provider, and specify any additional model parameters.
+
+| Field                     | Required | Description                                                                                                                |
```
I've rejigged the note around and added some info about this - should suffice for now I think?

mesellings reviewed (dismissed):
Assuming @maff is happy with the latest changes, I'm happy to approve and merge this for alpha-5 - great work on this @maff 🚀 👏

maff commented on docs/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent-example.md:
```diff
@@ -0,0 +1,236 @@
+- Instead of the user task, you could also use another LLM connector to verify the response of the AI Agent. For an example of this pattern, see the [fraud detection example](https://github.com/camunda/connectors/tree/main/connectors/agentic-ai/examples/fraud-detection).
+  :::
+
+## Tool Resolution
```
Sure, sounds great!
